bigbluebutton-html5/imports/ui/components/lock-viewers/context/context.js

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+export function DataStruct() {
+  return ({
+    isLocked: false,
+    lockSettings: {
+      disableCam: false,
+      disableMic: false,
+      disableNote: false,
+      disablePrivateChat: false,
+      disablePublicChat: false,
+      lockOnJoin: true,
+      lockOnJoinConfigurable: false,
+      lockedLayout: false,
+    },
+    userLocks: {
+      userWebcam: false,
```

Comment:
Why do we need to have individual props for userLocks? I thought this is to handle the un-locking of individual user but this could be represented with a single flag `userLock: false` rather than `userLock: { _ : false }

Reply:
The `isLocked` is the reference of the `mappedUser.isLocked`, My idea is avoid a lot of `user.isLocked && lockSetting` comparations, so I pass this consolidated.

Reply:
Could you not just use getters in this object so that the comparison is done in one spot? Seems better than duplicating the values.